src/components/apply/form/application-form.tsx

```diff
@@ -1,4 +1,133 @@
-// TODO: finish form component
+import type { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+} from "~/components/ui/form";
+import { Textarea } from "~/components/ui/textarea";
+import { api } from "~/utils/api";
+import { useAutoSave } from "~/components/hooks/use-auto-save";
+import { applicationStepSaveSchema } from "~/schemas/application";
+
 export function ApplicationForm() {
-  return <></>;
+  const utils = api.useUtils();
+  const { data: defaultValues } = api.application.get.useQuery();
+  const { mutate } = api.application.save.useMutation({
+    onSuccess: () => {
+      return utils.application.get.invalidate();
+    },
+  });
+
+  const form = useForm<z.infer<typeof applicationStepSaveSchema>>({
+    resolver: zodResolver(applicationStepSaveSchema),
+    mode: "onBlur",
+  });
+
+  useAutoSave(form, onSubmit, defaultValues);
+
+  function onSubmit(data: z.infer<typeof applicationStepSaveSchema>) {
+    mutate({
+      ...defaultValues,
+      ...data,
+    });
+  }
+
+  const maxChars = 150;
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
+        <div className="flex w-full flex-wrap gap-2">
+          <FormLabel className="w-full">
+            Tell us why you think you will win Hack Western 11.
+          </FormLabel>
+          <FormField
+            control={form.control}
+            name="question1"
+            render={({ field }) => (
+              <FormItem className="min-w-48 flex-1">
+                <FormControl>
+                  <Textarea
+                    {...field}
+                    value={field.value ?? ""}
+                    maxLength={maxChars}
+                    placeholder="Type your message here"
+                    variant="primary"
+                    onChange={(e) => {
+                      field.onChange(e);
+                    }}
+                  />
+                </FormControl>
+                <div className="text-sm text-gray-500">
+                  {maxChars - (field.value?.length ?? 0)} characters remaining
```

Comment:
OH I just realized, this shouldn't be characters. Check the design

<img width="623" alt="image" src="https://github.com/user-attachments/assets/09e3d1df-ba24-4442-be38-90af19c29dd2">

Reply:
With the current behaviour, negative remaining characters is possible which is a bit weird lol